_posts/2021-11-19-connecting-java-high-level-rest-client-with-opensearch-over-https.markdown

```diff
@@ -0,0 +1,176 @@
+```
+wget https://artifacts.opensearch.org/releases/bundle/opensearch/1.2.0/opensearch-1.2.0-linux-x64.tar.gz
+```
+
+Next, unzip the downloaded `tar.gz` file.
+```
+tar -xf opensearch-1.2.0-linux-x64.tar.gz
+```
+
+The OpenSearch distribution comes pre-installed with the security plugin. As part of the initial setup, you are required to setup the SSL certificates for encrypting the client to node and the node to node communication channels. The distribution comes with a tool for setting up demo certificates for a quick getting started experience. However, it is highly recommended to use certificates from a trusted [Certification Authority (CA)](https://en.wikipedia.org/wiki/Certificate_authority). You could also setup self-signed certificates by following this [documentation](https://opensearch.org/docs/latest/security-plugin/configuration/generate-certificates/). For the purpose of this blog, I have used the demo certificates available with the [install_demo_configuration.sh](https://github.com/opensearch-project/security/blob/main/tools/install_demo_configuration.sh) tool. 
+```
+cd opensearch-1.2.0/
+
+# Provide executable permissions to the tool if missing 
+chmod +x plugins/opensearch-security/tools/install_demo_configuration.sh
+
+# Install demo ssl certificates with install_demo_configurations.sh tool
+./plugins/opensearch-security/tools/install_demo_configurations.sh -y
+```
+
+Once the certificates are setup, increase the default `max_map_count` limit and start the OpenSearch cluster.  
+
+```
+# Increase mmap count limit 
+sudo sysctl -w vm.max_map_count=262144
+
+# To start the OpenSearch cluster
+bin/opensearch
+```
+
+At this point, your server is ready. You can verify using the below `curl` command. 
+```
+curl -k -XGET https://localhost:9200 -u 'admin:admin'
+```
+
+You should see a response similar to -
+
+```
+{
+  "name" : "smoketestnode",
+  "cluster_name" : "opensearch",
+  "cluster_uuid" : "2oMfOxJKQcmrDZ1e0OgXKw",
+  "version" : {
+    "distribution" : "opensearch",
+    "number" : "1.1.0",
+    "build_type" : "tar",
+    "build_hash" : "15e9f137622d878b79103df8f82d78d782b686a1",
+    "build_date" : "2021-10-04T21:29:03.079792Z",
+    "build_snapshot" : false,
+    "lucene_version" : "8.9.0",
+    "minimum_wire_compatibility_version" : "6.8.0",
+    "minimum_index_compatibility_version" : "6.0.0-beta1"
+  },
+  "tagline" : "The OpenSearch Project: https://opensearch.org/"
+}
+```
+
+Now that you have setup the OpenSearch server, let us move on to setting up the client.
+
+## Setup Java High Level REST Client
+
+The OpenSearch Java High Level REST Client is available on [Maven Central](https://search.maven.org/search?q=a:opensearch-rest-high-level-client). To start using, you need to add it as a dependency to your Java application. 
```

Comment:
to start using _it_, but you can also just get rid of that.

Add java-rest-client as a dependency to your Java application.